File: backend/app/schemas/debt.py
from datetime import datetime
from decimal import Decimal
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.employee import EmployeeRead
from app.schemas.product import ProductRead


DebtStatus = Literal[
    "active",
    "paid",
]


class DebtCreate(BaseModel):
    employee_id: UUID
    product_id: UUID

    quantity: Decimal = Field(
        gt=0,
        decimal_places=3,
        examples=[1],
    )

    note: str | None = Field(
        default=None,
        max_length=2000,
    )


class DebtUpdate(BaseModel):
    quantity: Decimal | None = Field(
        default=None,
        gt=0,
        decimal_places=3,
    )

    note: str | None = Field(
        default=None,
        max_length=2000,
    )


class DebtRead(BaseModel):
    model_config = ConfigDict(
        from_attributes=True
    )

    id: UUID
    employee_id: UUID
    product_id: UUID

    quantity: Decimal
    unit_price: Decimal
    total_amount: Decimal

    status: DebtStatus
    note: str | None

    created_at: datetime
    updated_at: datetime
    paid_at: datetime | None

    employee: EmployeeRead
    product: ProductRead

File: frontend/src/types/debt.ts
import type { Employee } from './employee'
import type { Product } from './product'

export type DebtStatus =
  | 'active'
  | 'paid'

export interface Debt {
  id: string
  employee_id: string
  product_id: string

  quantity: string
  unit_price: string
  total_amount: string

  status: DebtStatus
  note: string | null

  created_at: string
  updated_at: string
  paid_at: string | null

  employee: Employee
  product: Product
}

export interface DebtCreate {
  employee_id: string
  product_id: string
  quantity: number
  note: string | null
}

export interface DebtUpdate {
  quantity?: number
  note?: string | null
}